### DISCUSSION TITLE
rc.76 and "XXX_FILE" env variables

### DISCUSSION THREAD

**Discussion by gianpesto** [directus/directus#6314] [Feature Requests]
i'm running directus v9 on a jelastic cloudservice, node 14.17.0

So i went through a small adventure here, i thought it would be a nice idea to update directus as all of the sudden i had an error that said "Failed to load file from APP_FILE with value..." thrown by directus/dist/env.js .

Apparently this is a new feature introduced in rc.75 i guess:

"File Based Environment Variables (Docker Secrets)
Any of the environment variable values can be imported from a file, by appending _FILE to the environment variable name. For example: DB_PASSWORD_FILE="/run/secrets/db_password".
This is especially useful when used in conjunction with Docker Secrets, so you can keep sensitive data out of your compose files." (From [https://docs.directus.io/reference/environment-variables/](https://docs.directus.io/reference/environment-variables/))

this is fine, BUT my problem was that jelastic cloud service would have these unremovable env vars with a "_FILE" ending that made it impossible to use it unless i would be so bold changing the node_modules.

I know this might be very specific to this cloud-service, however anyone else might happen upon this problem too, if the env is filled with vars from somewhere else.

it might be a good idea to make the `key.endsWith('_FILE')` be `key.endsWith('_DIRECTUS_FILE')` or so...

**Comment by rijkvanzanten** [comment]
> this is fine, BUT my problem was that jelastic cloud service would have these unremovable env vars with a "_FILE" ending

Do you have an example of (some of) these env vars?

**Reply by gianpesto** [reply]
sure, in jelastic cloud you would have a non-removable env var called "APP_FILE" that would represent the entrypoint for your app, or the env var "PROCESS_MANAGER_FILE" that would represent the entrypoint for your process manager.

**Reply by rijkvanzanten** [reply]
I'd rather not start prefixing every env var if we don't absolutely have to. 

@paescuj I think an easy optimization we should do is to only load env vars from a whitelist of env vars that we actually use in Directus. While it doesn't solve for naming conflicts in env vars in general (though that's arguably very much based on where/how you run Directus in the first place), it would solve for this particular instance. Would you agree?

**Reply by paescuj** [reply]
> I'd rather not start prefixing every env var if we don't absolutely have to.

My immediate thought was the same. But after thinking about this for some time, this might be not even that bad. A lot of other applications do have prefixes on environment variables and other users might end up with similar conflicts like @gianpesto has right now.
However, I know this would be a big change and might have other disadvantages... What do you think @rijkvanzanten?

> @paescuj I think an easy optimization we should do is to only load env vars from a whitelist of env vars that we actually use in Directus. While it doesn't solve for naming conflicts in env vars in general (though that's arguably very much based on where/how you run Directus in the first place), it would solve for this particular instance. Would you agree?

I agree, that this would be a good solution as well!

**Reply by rijkvanzanten** [reply]
I don't think we have to prefix all the env vars. There are a couple ways to get around any potential naming conflicts already. You could run directus from a separate user, rely on a containerized environment (like docker), or not have Directus rely on environment variables in the first place, by relying on a configuration file 🤔

**Reply by paescuj** [reply]
> I don't think we have to prefix all the env vars. There are a couple ways to get around any potential naming conflicts already. You could run directus from a separate user, rely on a containerized environment (like docker), or not have Directus rely on environment variables in the first place, by relying on a configuration file 🤔

That's true! Actually, in a containerized environment you shouldn't run other apps (and thus have other environment variables) anyway 😄 
As I wrote, the "whitelist-solution" would be a good thing, IMO. Just wanted to ask you about your opinion on prefixes 😃

**Reply by Toilal** [reply]
SSL_CERT_FILE is a quite common environment variable too, used by OpenSSL to locate a custom CA certificate certificate file.

**Reply by denys-zarubin** [reply]
What about if make enntrypoint where to read EXAMPLE_FILE path and convert it not EXAMPLE env?

**Reply by paescuj** [reply]
> What about if make enntrypoint where to read EXAMPLE_FILE path and convert it not EXAMPLE env?

There is already a PR which is going to solve this issue: #13274

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
